src/server/http/version_check.js

```diff
@@ -0,0 +1,27 @@
+import { badRequest } from 'boom';
+
+export default function (kbnServer, server, config) {
+  const versionHeader = 'kbn-version';
+  const actualVersion = config.get('pkg.version');
+  const versionCheckDisabled = config.get('server.disableVersionCheck');
+  const xsrfProtectionDisabled = config.get('server.xsrf.disableProtection');
+  // this accepts the xsrf config strictly for legacy reasons
+  const disabled = versionCheckDisabled || xsrfProtectionDisabled;
```

Comment:
Yeah, don't review this PR right now.  The 4.6 PR changed a bunch of this and was merged this morning: https://github.com/elastic/kibana/pull/8152

I'll forward port those changes to this PR before we review again.